jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -217,225 +206,174 @@ public void execute(Runnable task) throws RejectedExecutionException
+                    while (task == null)
                     {
-                        long now = NanoTime.now();
-                        long lastEmpty = _lastEmptyNanoTime.get();
-                        if (size > 0 && _idleTimeNanos < NanoTime.elapsed(lastEmpty, now) && _lastEmptyNanoTime.compareAndSet(lastEmpty, now))
-                        {
-                            // it has been too long since we hit zero reserved threads, so are "busy" idle
-                            next = State.IDLE;
-                        }
-                        else
+                        if (!isRunning())
+                            return;
+
+                        // shrink if we are already removed or there are other reserved threads.
+                        // There is a small chance multiple threads will shrink below minSize
+                        if (getAvailable() > _minSize && _threads.remove(this, slot))
                         {
-                            // We will become a reserved thread if we can update the count below.
-                            next = State.RESERVED;
-                            size++;
+                            if (LOG.isDebugEnabled())
+                                LOG.debug("{} reservedThread shrank {}", ReservedThreadExecutor.this, this);
+                            return;
                         }
+                        task = waitForTask();
                     }
 
-                    // Update count for pending and size
-                    if (!_count.compareAndSet(count, pending, size))
-                        continue;
-
-                    if (LOG.isDebugEnabled())
-                        LOG.debug("{} was={} next={} size={}+{} capacity={}", this, _state, next, pending, size, _capacity);
-                    _state = next;
-                    if (next != State.RESERVED)
-                        break;
-
-                    // We are reserved whilst we are waiting for an offered _task.
-                    Runnable task = reservedWait();
-
-                    // Is the task the STOP poison pill?
-                    if (task == STOP)
-                        break;
-
-                    // Run the task
                     try
                     {
-                        _state = State.RUNNING;
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("{} reservedThread run {} on {}", ReservedThreadExecutor.this, task, this);
                         task.run();
                     }
-                    catch (Throwable e)
-                    {
-                        LOG.warn("Unable to run task", e);
-                    }
-                    finally
+                    catch (Throwable t)
                     {
-                        // Clear any interrupted status.
-                        Thread.interrupted();
+                        if (LOG.isDebugEnabled())
+                            LOG.debug("reserved error", t);
```

Comment:
Restore the old log, this is for exception thrown by the task.

```suggestion
                        if (LOG.isDebugEnabled())
                            LOG.debug("{} task {} failure", ReservedThreadExecutor.this, task, t);
```